lib/expect.h

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef EXPECT_H_
+#define EXPECT_H_
+
+#include <complex>
+
+#include "fuser.h"
+#include "gate_appl.h"
+
+namespace qsim {
+
+template <typename Gate>
+struct OpString {
+  std::complex<double> weight;
+  std::vector<Gate> ops;
+};
+
+/**
+ * Computes the expectation value of the sum of operator strings. Operators
+ * can be any supported gates. Uses a temporary state vector.
+ * @param param Options for gate fusion.
+ * @param strings Operator strings.
```

Comment:
What is an "operator string" in this context? Could we include this info in the docstring? (Same for the other overload of this method)

Reply:
It is just a sequence of operators. I updated the docstring.